src/main/java/com/devathon/preguntonic/controllers/GameControllerImpl.java

```diff
@@ -0,0 +1,279 @@
+package com.devathon.preguntonic.controllers;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+
+import com.devathon.preguntonic.constants.Constants;
+import com.devathon.preguntonic.domain.Question;
+import com.devathon.preguntonic.dto.GameStatusDto;
+import com.devathon.preguntonic.dto.PlayerQuestionResponseDto;
+import com.devathon.preguntonic.dto.PlayerStatusDto;
+import com.devathon.preguntonic.dto.QuestionDto;
+import com.devathon.preguntonic.dto.QuestionResultDto;
+import com.devathon.preguntonic.model.Game;
+import com.devathon.preguntonic.model.GameStatus;
+import com.devathon.preguntonic.model.Player;
+import com.devathon.preguntonic.model.PlayerStatus;
+import com.devathon.preguntonic.model.Room;
+import com.devathon.preguntonic.model.RoomStatus;
+import com.devathon.preguntonic.services.DummyRoomService;
+import com.devathon.preguntonic.services.QuestionService;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Controller
+public class GameControllerImpl implements GameController {
+
+    @Autowired
+    private DummyRoomService roomService;
+
+    @Autowired
+    private QuestionService questionService;
+    
+    @Override
+    public void gameSubscription(final String code) {
+
+        // Check if the room exists
+        Optional<Room> optRoom = roomService.getRoom(code);
+        if (optRoom.isEmpty()) {
+            log.error("Room {} not found", code);
+            return;
+        }
+
+        log.info("Game subscription for room {}", code);
+    }
+
+    @Override
+    public GameStatusDto joinGame(final String code, final UUID playerId) {
+
+        // Check if the room exists
+        Optional<Room> optRoom = roomService.getRoom(code);
+        if (optRoom.isEmpty()) {
+            log.error("Room {} not found", code);
+            return null;
+        }
+
+        Room room = optRoom.get();
+
+        // Check if the player is already in the room
+        if (!room.getPlayers().containsKey(playerId)) {
+            log.error("Player {} is not in the room {}", playerId, code);
+            return null;
+        }
```

Comment:
I think we need in Room service a getPlayers Method that handle this validations I can add it in this PR https://github.com/Devathon-2024-team5/Preguntonic-backend/pull/16/

But do you prefer that the method return an Exception or null if Room or player not found? For now I prefer Exception but WDYT?

Reply:
Also maybe you can use the current `Optional<BasicPlayer> getPlayer(String roomCode, int playerId)` maybe should i change the returned object?

Reply:
We have to coordinate to do that. We can use room service methods as well. About exception... it is a good idea, but, how do we could return this exceptions to the frontend?? For this reason i prefer an object that is always returned and filled with the good/bad results